libsolidity/formal/ExpressionFormatter.cpp

```diff
@@ -245,7 +245,14 @@ std::optional<std::string> expressionToString(smtutil::Expression const& _expr,
 	if (smt::isNumber(*_type))
 	{
 		solAssert(_expr.sort->kind == Kind::Int);
-		solAssert(_expr.arguments.empty());
+		solAssert(_expr.arguments.empty() || _expr.name == "-", "");
```

Comment:
Is this related to Z3 or just an improvement overall?
Also, since the assert error message is empty, we can omit it.
```suggestion
		solAssert(_expr.arguments.empty() || _expr.name == "-");
```

Reply:
This was a bug that was dormant and was exposed by the switch to using SMT-LIB interface much more.

I've removed the unnecessary second argument now.